### DISCUSSION TITLE
Check if UniqueEntity exists?

### DISCUSSION THREAD

**Discussion by Danon5** [friflo/Friflo.Engine.ECS#9] [Q&A]
Is there a built-in API (besides manually iterating) to check if a UniqueEntity has been created or not? Would be nice to see a HasUniqueEntity and TryGetUniqueEntity in the default API since GetUniqueEntity simple throws an exception if it doesn't exist and there is seemingly no built-in way to check for it. Making custom extension methods that iterate and compare strings isn't difficult but feels a bit strange, although maybe that's all that the built-in API is doing with GetUniqueEntity?

**Comment by friflo** [comment]
As you mentioned - for now the property `EntityStore.UniqueEntities` returns all unique entities.
The returned entities can be iterated to find a specific entity by a string identifier.

Using this mechanism show the runtime complexity of this approach caused when iteration the result.
The intention of `EntityStore.GetUniqueEntity(string)` is to call this method only once and used the returned entity in subsequent code.
